tests/sles4sap/redirection_tests/ensa2_hanadb_sapcontrol_move_ascs.pm

```diff
@@ -0,0 +1,117 @@
+    while ($outputs = sapcontrol(webmethod => 'HACheckConfig', instance_id => $instance_id, sidadm => $sidadm, return_output => 1)) {
+        last unless ($outputs =~ /ERROR|FAIL/);
+        record_info("ERROR found in HACheckConfig: $outputs", "sleep 30s and try again");
+        sleep 30;
+        $looptime -= 30;
+        last if ($looptime <= 0);
+    }
+    sapcontrol(webmethod => 'HACheckFailoverConfig', instance_id => $instance_id, sidadm => $sidadm);
+}
+
+sub execute_ascs_failover {
+    my (%args) = @_;
+    my $id = $args{instance_id};
+    my $user = $args{instance_user};
+    my $instance_hostname_scs = $args{instance_hostname_scs};
+    my $instance_hostname_ers = $args{instance_hostname_ers};
+    my $wait_for_target = $args{wait_for_target};
+
+    # Execute web method checks
+    webmethod_checks($id, $user);
+    record_info('Failover', "Executing 'HAFailoverToNode'. Failover from $instance_hostname_ers to $instance_hostname_scs");
+    sapcontrol(webmethod => 'HAFailoverToNode', instance_id => $id, additional_args => "\"\"");
+
+    # Wait for failover to finish and check resource locations
+    record_info('Fail wait', 'Waiting for failover to complete');
+    my $sid = get_required_var('SAP_SID');
+    crm_check_resource_location(resource => "g-${sid}_ASCS", wait_for_target => $wait_for_target, timeout => 360);
+}
+
+sub run {
+    my ($self, $run_args) = @_;
+    my %redirection_data = %{$run_args->{redirection_data}};
+
+
+    # Connect to deployer VM (by default) to load env variables and get get instance IDs
+    connect_target_to_serial();
+    load_os_env_variables();
+    my $instance_id_scs = get_sdaf_instance_id(pattern => 'SCS');
+    my $instance_id_ers = get_sdaf_instance_id(pattern => 'ERS');
+    # Disonnect from deployer VM
+    disconnect_target_from_serial();
+
+    # Get scs instance data
+    my %instance_data_scs = %{$run_args->{redirection_data}{nw_ascs}};
+    my $instance_hostname_scs = (keys(%instance_data_scs))[0];
+    my $ip_scs = $instance_data_scs{$instance_hostname_scs}{ip_address};
+    my $user_scs = $instance_data_scs{$instance_hostname_scs}{ssh_user};
+    # Get ers instance data
+    my %instance_data_ers = %{$run_args->{redirection_data}{nw_ers}};
+    my $instance_hostname_ers = (keys(%instance_data_ers))[0];
+    my $ip_ers = $instance_data_ers{$instance_hostname_ers}{ip_address};
+    my $user_ers = $instance_data_ers{$instance_hostname_ers}{ssh_user};
+
+    # Connect to scs VM
+    connect_target_to_serial(destination_ip => $ip_scs, ssh_user => $user_scs, switch_root => 1);
+    # Check for idle cluster and no failed resources
+    record_info('Cluster wait', 'Waiting for resources to start');
+    wait_until_resources_started();
+    wait_for_idle_cluster();
+    record_info('Cluster check', 'Checking state of cluster resources');
+    check_cluster_state();
+
+    my $topology = calculate_hana_topology(input => script_output('SAPHanaSR-showAttr --format=script'));
```

Comment:
I think the topology collection is not needed. 
If you check the VR it does not output anything really:
https://openqaworker15.qa.suse.cz/tests/320081#step/ensa2_hanadb_sapcontrol_move_ascs/84

With that import at L21 won't be needed either.

Reply:
> I think the topology collection is not needed. If you check the VR it does not output anything really: https://openqaworker15.qa.suse.cz/tests/320081#step/ensa2_hanadb_sapcontrol_move_ascs/84
> 
> With that import at L21 won't be needed either.

Deleted.